code/_onclick/hud/parallax.dm

```diff
@@ -0,0 +1,254 @@
+		var/world_time = world.time
+		last_delay = world_time - C.last_parallax_shift
+		last_delay = min(last_delay, 2)
+		C.last_parallax_shift = world_time
+
+	for(var/obj/screen/parallax_layer/L in C.parallax_layers)
+		L.offset_x -= offset_x * L.speed
+		L.offset_y -= offset_y * L.speed
+
+		if(C.looping_mode == 0)
+			if(L.offset_x > 240)
+				L.offset_x -= 480
+			if(L.offset_x < -240)
+				L.offset_x += 480
+			if(L.offset_y > 240)
+				L.offset_y -= 480
+			if(L.offset_y < -240)
+				L.offset_y += 480
+		else if(C.looping_mode == 1)
+			if(L.offset_x > 0)
+				L.offset_x -= 480
+			if(L.offset_x < -480)
+				L.offset_x += 480
+			if(L.offset_y > 0)
+				L.offset_y -= 480
+			if(L.offset_y < -480)
+				L.offset_y += 480
+		else if(C.looping_mode == 2)
+			if(L.offset_x >= 480)
+				L.offset_x -= 480
+			if(L.offset_x < 0)
+				L.offset_x += 480
+			if(L.offset_y >= 480)
+				L.offset_y -= 480
+			if(L.offset_y < 0)
+				L.offset_y += 480
+
+		if(C.do_smoothing && (offset_x != 0 || offset_y != 0) && (offset_x == 1 || offset_x == -1 || offset_y == 1 || offset_y == -1))
+			L.transform = matrix(1, 0, offset_x*L.speed, 0, 1, offset_y*L.speed)
+			animate(L, transform=matrix(), time = last_delay, flags = ANIMATION_END_NOW)
+
+		L.screen_loc = "CENTER-7:[L.offset_x],CENTER-7:[L.offset_y]"
+
+// Plays the launch animation for parallax
+/datum/hud/proc/parallax_launch_anim(dir = 4, slowing = 0)
+	var/client/C = mymob.client
+	C.do_smoothing = 0
+	if(dir == 4 || dir == 1)
+		C.looping_mode = 1
+	else
+		C.looping_mode = 2
+	for(var/obj/screen/parallax_layer/L in C.parallax_layers)
+		animate(L) // Cancel the current animation
+		var/M = L.speed * 240
+		var/O = -480 + M
+		switch(dir)
+			if(1)
+				L.transform = matrix(1, 0, 0, 0, 1, M)
+				L.offset_y += O
+			if(2)
+				L.transform = matrix(1, 0, 0, 0, 1,-M)
+				L.offset_y -= O
+			if(4)
+				L.transform = matrix(1, 0, M, 0, 1, 0)
+				L.offset_x += O
+			if(8)
+				L.transform = matrix(1, 0,-M, 0, 1, 0)
+				L.offset_x -= O
+	update_parallax() // Adjust the layers, they should all now be in the appropriate corner.
+	for(var/obj/screen/parallax_layer/L in C.parallax_layers)
+		animate(L, transform = matrix(), time = 50, easing = QUAD_EASING | (slowing ? EASE_OUT : EASE_IN), flags = ANIMATION_END_NOW)
+	spawn(50)
+		if(slowing)
+			C.do_smoothing = 1
+			C.looping_mode = 0
+
+
+// Helper global procs for performing shuttle animations
+/proc/parallax_launch_in_area(var/area/A, dir = 4, slowing = 0)
+	for(var/mob/M in mob_list)
+		if(M.client && M.hud_used && get_area(M) == A)
+			M.hud_used.parallax_launch_anim(dir, slowing)
+
+/proc/parallax_movedir_in_area(var/area/A, dir = 4)
+	A.parallax_movedir = dir
+	for(var/mob/M in mob_list)
+		if(M.client && M.hud_used && get_area(M) == A)
+			M.hud_used.update_parallax()
+
+/datum/hud/proc/update_parallax_movingmob()
+	var/client/C = mymob.client
+	var/atom/movable/A = C.eye
+	if(!A)
+		return
+	while(istype(A.loc, /atom/movable))
+		A = A.loc
```

Comment:
MrPerson most likely thinks this is a get_turf, but its not exactly that, its looking for one layer before the turf, ie, the movable that is on the turf. We don't have a helper for that.

but, we do have a helper for ismovable you should use.